examples/flow-counter/actions/counter.js

```diff
@@ -0,0 +1,39 @@
+/* @flow */
+
+import type { Action, AppState } from '../types';
+import type { Thunk } from 'redux-thunk';
+
+export const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
+export const DECREMENT_COUNTER = 'DECREMENT_COUNTER';
+
+export function increment() : Action {
+  return {
+    type: INCREMENT_COUNTER
+  };
+}
+
+export function decrement() : Action {
+  return {
+    type: DECREMENT_COUNTER
+  };
+}
+
+export function incrementIfOdd() : Thunk<AppState, Action> {
+  return (dispatch, getState) => {
+    let { counter } = getState();
```

Comment:
I changed this `const` to a `let` because of this bug in Flow, that is now fixed in master: https://github.com/facebook/flow/issues/844